crates/cli/src/templates/binding.java

```diff
@@ -0,0 +1,62 @@
+package PARSER_NS_CLEANED.jtreesitter.LOWER_PARSER_NAME;
+
+import java.lang.foreign.*;
+
+public final class PARSER_CLASS_NAME {
+    private static final ValueLayout VOID_PTR =
+            ValueLayout.ADDRESS.withTargetLayout(MemoryLayout.sequenceLayout(Long.MAX_VALUE, ValueLayout.JAVA_BYTE));
+    private static final FunctionDescriptor FUNC_DESC = FunctionDescriptor.of(VOID_PTR);
+    private static final Linker LINKER = Linker.nativeLinker();
+    private static final TreeSitterJava INSTANCE = new PARSER_CLASS_NAME();
+
+    private final Arena arena = Arena.ofAuto();
+    private SymbolLookup lookup = null;
+
+    /**
+     * Get the tree-sitter language for this grammar.
+     */
+    public static MemorySegment language() {
+        if (INSTANCE.lookup == null)
+            INSTANCE.lookup = INSTANCE.findLibrary();
+        return language(INSTANCE.lookup);
+    }
+
+    /**
+     * Get the tree-sitter language for this grammar.
+     *
+     * <strong>The {@linkplain Arena} used in the {@code lookup}
+     * must not be closed while the language is being used.</strong>
+     */
+    public static MemorySegment language(SymbolLookup lookup) {
+        return call(lookup, "tree_sitter_PARSER_NAME");
+    }
+
+    private SymbolLookup findLibrary() {
+        try {
+            var library = System.mapLibraryName("tree-sitter-KEBAB_PARSER_NAME");
+            return SymbolLookup.libraryLookup(library, arena);
+        } catch (IllegalArgumentException ex1) {
+            try {
+                return SymbolLookup.loaderLookup();
+            } catch (UnsatisfiedLinkError ex2) {
+                ex1.addSuppressed(ex2);
+                throw ex1;
+            }
+        }
+    }
+
+    private static UnsatisfiedLinkError unresolved(String name) {
+        return new UnsatisfiedLinkError("Unresolved symbol: %s".formatted(name));
+    }
+
+    @SuppressWarnings("SameParameterValue")
+    private static MemorySegment call(SymbolLookup lookup, String name) throws UnsatisfiedLinkError {
```

Comment:
What about using `Language#load(SymbolLookup, String)` from jtreesitter instead? I know that you originally wanted to get rid of it again in the future, but I still think it is quite useful and could also avoid the code duplication here.

Reply:
I have many issues with that function.
Most importantly, I don't want a public function that can load arbitrary symbols.